### PR TITLE
webhooks: Add `head_commit` field to push payload.

### LINKED ISSUE #5266: include head_commit in push webhooks if tag
It would be great if Gitea could include the `head_commit` in the `push` webhook if the object is a tag. For reference this is what GitHub sends in the push hook if the object is a tag:

```
{
  "ref": "refs/tags/v0.0.1",
  "before": "0000000000000000000000000000000000000000",
  "after": "d2b75aa7797ec26b088fa2dd527e9d2c052fcedd",
  "created": true,
  "deleted": false,
  "forced": false,
  "base_ref": "refs/heads/master",
  "compare": "https://github.com/bradrydzewski/drone-test-go/compare/v0.0.1",
  "commits": [

  ],
  "head_commit": {
    "id": "d2b75aa7797ec26b088fa2dd527e9d2c052fcedd",
    "tree_id": "f38d6b02c9ee18de68e5e6682721a22df4c9a29d",
    "distinct": true,
    "message": "Update .drone.yml",
    "timestamp": "2018-06-19T19:03:12-07:00",
    "url": "https://github.com/bradrydzewski/drone-test-go/commit/d2b75aa7797ec26b088fa2dd527e9d2c052fcedd",
    "author": {
      "name": "Brad Rydzewski",
      "email": "noreply@github.com",
      "username": "bradrydzewski"
    },
    "committer": {
      "name": "GitHub",
      "email": "noreply@github.com",
      "username": "web-flow"
    },
    "added": [

    ],
    "removed": [

    ],
    "modified": [
      ".drone.yml"
    ]
  }
}
```

For my specific use case, I would like to display the commit details (email address, commit message, etc) in the user interface in [Drone](https://github.com/drone/drone). Currently these fields are blank. Thanks!

### PR DESCRIPTION
**Pull request by HarshitOnGitHub**
Would appreciate comments/feedback on whether the approach taken is correct or not. Have tested the changes manually.

Fixes: #5266.

### CODE REVIEW

**Comment by HarshitOnGitHub** [comment]
(Updated dependency, build should pass now!)

**Comment by lafriks** [comment]
Would be great if convert.ToCommit could be reused somehow to get api.PayloadCommit to not have so much duplicated code.

**Comment by lafriks** [comment]
@HarshitOnGitHub can you fix to reuse `convert.ToCommit`?

**Comment by HarshitOnGitHub** [comment]
@lafriks `convert.ToCommit` can't be used directly as it causes cyclic dependency. I was thinking of moving the function to some other place but couldn't decide on a good new place for it.

**Comment by HarshitOnGitHub** [comment]
@lafriks Thoughts?

**Comment by HarshitOnGitHub** [comment]
Any help/suggestions would be appreciated! :)

File: models/action.go [hunk 1]
```diff
@@ -720,6 +781,7 @@ func CommitRepoAction(opts CommitRepoActionOptions) error {
 			After:      opts.NewCommitID,
 			CompareURL: setting.AppURL + opts.Commits.CompareURL,
 			Commits:    opts.Commits.ToAPIPayloadCommits(repo.HTMLURL()),
+			HeadCommit: getHeadCommit(repo, opts.NewCommitID),
```
Comment (lunny) [inline models/action.go:784]: Could we get the head PayloadCommit from APIPayLoadCommits ?

**Review by lunny — CHANGES_REQUESTED** [review]

**Comment by lunny** [comment]
Please resolve the conflicts

**Pull request closed without merging**